src/StatsdClient/RandomGenerator.cs

```diff
@@ -4,15 +4,26 @@ namespace StatsdClient
 {
     public class RandomGenerator : IRandomGenerator
     {
-        readonly Random _random;
+        [ThreadStatic]
+        static Random _random;
+
+        private static Random Random
+        {
+            get
+            {
+                var random = _random;
+                if (random != null) return random;
+                return _random = new Random(Guid.NewGuid().GetHashCode());

```

Comment:
Interesting, never ran across this. Good to know! OK, I'll go ahead and merge. Thanks for clearing that up :)